pkg/test/framework/components/istio/operator.go

```diff
@@ -730,3 +743,132 @@ func deployCACerts(workDir string, env *kube.Environment, cfg Config) error {
+	if err != nil {
+		return err
+	}
+	eps := &kubeApiCore.Endpoints{
+		ObjectMeta: kubeApiMeta.ObjectMeta{
+			Name:      istiodSvcName,
+			Namespace: cfg.SystemNamespace,
+		},
+		Subsets: []kubeApiCore.EndpointSubset{
+			{
+				Addresses: []kubeApiCore.EndpointAddress{
+					{
+						IP: discoveryAddress,
+					},
+				},
+				Ports: []kubeApiCore.EndpointPort{
+					{
+						Name:     "tcp-istiod",
+						Protocol: kubeApiCore.ProtocolTCP,
+						Port:     15012,
+					},
+					{
+						Name:     "tcp-webhook",
+						Protocol: kubeApiCore.ProtocolTCP,
+						Port:     15017,
+					},
+				},
+			},
+		},
+	}
+
+	_, err = cluster.CoreV1().Endpoints(cfg.SystemNamespace).Create(context.TODO(), eps, kubeApiMeta.CreateOptions{})
+	if err != nil {
+		return err
+	}
+	err = retry.UntilSuccess(func() error {
+		_, err := cluster.CoreV1().Services(cfg.SystemNamespace).Get(context.TODO(), istiodSvcName, kubeApiMeta.GetOptions{})
+		if err != nil {
+			return err
+		}
+		_, err = cluster.CoreV1().Endpoints(cfg.SystemNamespace).Get(context.TODO(), istiodSvcName, kubeApiMeta.GetOptions{})
+		if err != nil {
+			return err
+		}
+		return nil
+	}, retry.Timeout(20*time.Second), retry.Delay(2*time.Second))
+	return err
+}
+
+// If we are under central Istiod which uses remote KUBECONFIG, we follow below install sequences:
+// install istiod Remote + base in control cluster first.
+// install istiod in remote cluster
+// get istiod loadbalancer address(discoveryAddress)
+// create istiod service and endpoints in control cluster
+func deployWithRemoteKubeConfig(iopFile, remoteIopFile string, env *kube.Environment, cfg Config, i *operatorComponent) error {
+	var err error
+	for _, cluster := range env.KubeClusters {
+		if env.IsControlPlaneCluster(cluster) {
+			cluster := cluster
+			if err := deployControlPlane(i, cfg, cluster, iopFile); err != nil {
+				return fmt.Errorf("failed deploying control plane to cluster %s: %v", cluster.Name(), err)
+			}
+			break
+		}
+	}
+	var istiodAddress net.TCPAddr
+	for _, cluster := range env.KubeClusters {
+		if !env.IsControlPlaneCluster(cluster) {
+
+			if err := deployControlPlane(i, cfg, cluster, remoteIopFile); err != nil {
+				return fmt.Errorf("failed deploying control plane to cluster %s: %v", cluster.Name(), err)
+			}
+			err = retry.UntilSuccess(func() error {
+				istiodAddress, err = i.RemoteDiscoveryAddressFor(cluster)
+				if err != nil {
+					return err
+				}
+				return nil
+			}, retry.Timeout(90*time.Second), retry.Delay(2*time.Second))
+			if err != nil {
+				return err
+			}
+			err = patchIstiodCustomHost(i, cfg, cluster)
+			if err != nil {
+				return err
+			}
+			break
+		}
+	}
+
+	discoveryAddress := istiodAddress.IP.String()
+	scopes.Framework.Infof("discoveryAddress : %s", discoveryAddress)
+	for _, cluster := range env.KubeClusters {
+		if env.IsControlPlaneCluster(cluster) {
+			err = createIstiodEndpoint(discoveryAddress, cfg, cluster)
```

Comment:
I think if `values.global.remotePilotAddress` is specified in `RemoteClusterValues`, this will already be done for you

Reply:
The issue here is we need install istiodremote + base first, then istiod and let it point to the KUBECONFIG in the base cluster.

So we can not know the address before hand, this is the reason we need to get the address and create endpoints here later.

Reply:
Ah, sorry